### PR TITLE
Check Markdown files in tmt lint only when `manual=True`

### PR DESCRIPTION
**Pull request by dkarpele**
#994

`tmt lint` will check test files only when `manual` attribute is True.
The user complains that the exception `Install tmt-test-convert to export
tests.` shows in `tmt lint` although he doesn't want to export anything.
`return_markdown_file` method won't be used in `tmt lint`.
Tests updated.

Signed-off-by: Denis Karpelevich <dkarpele@redhat.com>

### CODE REVIEW

**Comment by dkarpele** [comment]
Hi @psss and @dennisbrendel, please share your thoughts.

**Comment by dennisbrendel** [comment]
@dkarpele thank you! The code looks fine to me, it works, and I think it catches all the error cases.

Note: I personally would think about refactoring lint() before adding more logic :-)

**Comment by psss** [comment]
The `/tests/plan/lint` failure on rawhide is expected (a known `fmf`/`git` issue).
Not sure why `/tests/run/worktree` is failing as well, investigating.

**Comment by psss** [comment]
> Not sure why `/tests/run/worktree` is failing as well, investigating.

Seems to be caused by `selinux` not allowing to access the test code. Can reproduce on latest rawhide with `setenforce 1` and works fine with `setenforce 0`.

**Comment by psss** [comment]
After discussion with @lukaszachy and @milosmalik filed a `container-selinux` bug [2044249](https://bugzilla.redhat.com/show_bug.cgi?id=2044249).

File: tmt/base.py [hunk 1]
```diff
@@ -475,14 +475,30 @@ def lint(self):
...
-            invalid_md_file = tmt.export.check_md_file_respects_spec(md_path)
-            if invalid_md_file:
-                for i in invalid_md_file:
-                    verdict(False, i)
-            else:
-                verdict(True, "correct headings are used in the Markdown file")
+        if self.manual:
+            try:
+                tests_amount = len(self.test.split())
+            except AttributeError:
```
Comment (psss) [inline tmt/base.py:481]: Not necessary. Each test has to have the `test` attribute defined otherwise it would not be detected as test.

**Comment by dkarpele** [comment]
This line covers the case:
```
test:
manual: true
```

If user defines this fmf config, `self.test` will be `None` and script will fail with `{AttributeError}'NoneType' object has no attribute 'split'`. So, I think, it makes sense to have this catch.

File: tmt/base.py (hunk 1 above)
Comment (dkarpele) [inline tmt/base.py:481]: This line covers the case:
```
test:
manual: true
```

If user defines this fmf config, `self.test` will be `None` and script will fail with `{AttributeError}'NoneType' object has no attribute 'split'`. So, I think, it makes sense to have this catch.

**Review by psss — APPROVED** [review]
Thanks for all the improvements, @dkarpele! Seems to be working fine. I just think that we should not allow `None` as a valid `test` attribute value. So I've added check for that as well. This also ensures that `test` will always be a string. I've squashed your changes, rebased on the latest master and simplified the tests a bit in 21962777. Please, have a look if everything looks good.

**Comment by dkarpele** [comment]
Thank you for the review and update @psss ,
LGTM

**Pull request merged**